jsonrpc-specs/src/schemas/discv5.json

```diff
@@ -0,0 +1,58 @@
+{
+  "Bucket": {
+    "title": "Bucket info",
+    "description": "List of up to 16 hex encoded nodeIds, ordered from least-recently connected to most-recently connected.",
+    "type": "array",
+    "items": { "$ref": "#/components/schemas/bytes32" }
+  },
+  "Enr": {
+    "title": "Base64 encoded ENR",
+    "type": "string",
+    "pattern": "^enr:[a-zA-Z0-9_:-]{179}$"
+  },
+  "kBuckets": {
+    "title": "kBuckets",
+    "description": "The buckets comprising the routing table.",
+    "type": "array",
+    "items": { "$ref": "#/components/schemas/Bucket" }
+  },
+  "KVpairs": {
+    "title": "kvPairs",
+    "description": "Key-value pairs",
+    "type": "object",
+    "properties": {
+      "id": {
+        "title": "Name of identity scheme, e.g. “v4”",
+        "type": "string"
+      },
+      "secp256k1": {
```

Comment:
For its usage in `discv5_updateNodeInfo` you can really update the `secp256k1` or `id` values as you are not passing any new private key.

Reply:
This makes it an array of these objects in `discv5_updateNodeInfo`, while it should probably just be an array of k:v pairs (without the schema)

Reply:
I'm changing `discv5_updateNodeInfo` to accept two params:

- socketAddr (ENR socket address)
- isTcp (boolean, TCP or UDP socket, not required, default value: false)

Example:
`{"jsonrpc":"2.0","id":1,"params":[socketAddr: "192.168.0.25:12345", isTcp: "true"],"method":"discv5_updateNodeInfo"}`

Let me know if this makes sense.

Reply:
It makes sense, but it might grow old if/when more fields require updates (Yes, we might not need that).

But perhaps it is just easier allow any key:value pair update in the ENR. Also, we need to provide the private key after all as a new signature will have to be made. (So in theory you could actually update the `secp256k1` field also, if a new private key is provided. In the Nim implementation this is currently not allowed though).

Reply:
Above statement is not really true, the private network key would typically be accessible in the implementation.

So basically:
- Either provide private network key also as parameter and allow all k:v pairs to change.
- Don't provide private network key as parameter and allow all but public key and id (error on those) to change, or as your current suggestion allow only the address.

Reply:
ok, for now, I'm going to leave it as it is, we may change this in the future based on our needs.